resources/benchmark-runner.mjs

```diff
@@ -145,6 +155,15 @@ class PageElement {
         const event = new contentWindow.MouseEvent(type, eventOptions);
         this.#node.dispatchEvent(event);
     }
+
+    querySelector(selector) {
+        const root = this.#node.shadowRoot ?? this.#node;
```

Comment:
if this node has a shadow root, how would you access children for this node then? I'd use an optional boolean parameter "useShadowRoot" to decide. This is a bit like `getComputedStyle` taking a parameter for pseudo elements.

Reply:
Or maybe Page's `querySelector` and `querySelectorAll` (and `getParent`) should be moved here, with Page's version of them defering to PageElement.

Reply:
@julienw - I made some updates - let me know if that's more what you had in mind

Reply:
My question was more: what happens if this element has both a shadowRoot and children? How do we decide if we go down the shadowRoot path or the in-the-main-document path? Maybe we really want 2 different functions here: querySelector vs querySelectorInShadowRoot?

But this is theoretical so we might as well decide when/if we need it later. What do the others think?

Reply:
I see what you mean. I am open to rename and keeping the original Page.querySelector / Page.querySelectorAll methods if that's preferred.

Reply:
I see what you are saying - at this point it's not possible to querySelector for a direct descendant if the node has a shadow root. I agree we could punt on this and figure it out when it's needed, but I also wonder if the existence of a non-empty path could be used as semantics to imply that you want to use shadow root.

So if you wanted to jump between shadow and non shadow you could first get to the element who's direct descendant you care about using a querySelector on the document with the path (which will prefer the shadow root). Then take a reference to that element and querySelector on that without a path (which will use the element and not the root). Then if you need to go further into a shadow tree you can querySelector again with a path. Or we could just create new methods.

Reply:
As soon as you want to query a wrapped node, you'd need to use PageElement.querySelector, which still prefers the shadowRoot. How about this for the PageElement methods?

```
querySelectorInShadowRoot(selector, path = []) {
        const lookupStartNode = this.#node.shadowRoot ?? this.#node;
        const element = getParent(lookupStartNode, path).querySelector(selector);

        if (element === null)
            return null;
        return new PageElement(element);
    }

    querySelector(selector) {
        const element = this.#node.querySelector(selector);

        if (element === null)
            return null;
        return new PageElement(element);
    }

```

Reply:
I was suggesting we use the existence of a non-empty path to in the same manner as for the document methods (prefer shadow root if there is >= 1 element in the array, and the element if not). But I really don't have a strong opinion on the API shape until if/when we need it